ra_aid/file_listing.py

```
@@ -95,42 +99,66 @@ def get_file_listing(
         FileListerError: For other unexpected errors
     """
     try:
-        # Check if directory is a git repo first
+        # Check if directory exists and is accessible
+        if not os.path.exists(directory):
+            raise DirectoryNotFoundError(f"Directory not found: {directory}")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)